yjit/src/options.rs

```diff
@@ -0,0 +1,124 @@
+use std::ffi::CStr;
+
+// Command-line options
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+#[repr(C)]
+pub struct Options {
+    // Size of the executable memory block to allocate in MiB
+    pub exec_mem_size : usize,
+
+    // Number of method calls after which to start generating code
+    // Threshold==1 means compile on first execution
+    pub call_threshold : usize,
+
+    // Generate versions greedily until the limit is hit
+    pub greedy_versioning : bool,
+
+    // Disable the propagation of type information
+    pub no_type_prop : bool,
+
+    // Maximum number of versions per block
+    // 1 means always create generic versions
+    pub max_versions : usize,
+
+    // Capture and print out stats
+    pub gen_stats : bool,
+
+    /// Dump compiled and executed instructions for debugging
+    pub dump_insns: bool,
+
+    /// Verify context objects (debug mode only)
+    pub verify_ctx: bool,
+
+    /// Whether or not to assume a global constant state (and therefore
+    /// invalidating code whenever any constant changes) versus assuming
+    /// constant name components (and therefore invalidating code whenever a
+    /// matching name component changes)
+    pub global_constant_state: bool,
+}
+
+// Initialize the options to default values
+pub static mut OPTIONS: Options = Options {
```

Comment:
I'd consider using something else other than a static mut for holding on to options. Especially a pub one. Is there any reason for this to be public? 
rust-lang/rust#53639